### PR TITLE
[Validator] Test sets of errors rather than strings

### PR DESCRIPTION
**Pull request by c-harding**
The test messages were annoying me, so I’ve switched to testing the sets of error strings rather than the resulting error message. Don’t worry, the old method is still tested as before

Turn off whitespace to shrink the diff to a more manageable +90–69

### CODE REVIEW

File: src/test/scala/temple/DSL/semantics/ValidatorTest.scala [hunk 1]
```diff
@@ -14,210 +14,230 @@ class ValidatorTest extends FlatSpec with Matchers {
...
-              metadata = Seq(Writable.All, Readable.This),
-            ),
-          ),
+        templefileWithUserAttributes(
+          "b" -> Attribute(FloatType(max = Some(0), min = Some(1))),
+          "c" -> Attribute(FloatType(precision = -1)),
         ),
       )
-    } should have message singleError("#writable(all) is not compatible with #readable(this) in User")
+    } should have message {
```
Comment (jaylees14) [inline src/test/scala/temple/DSL/semantics/ValidatorTest.scala:236]: I will never stop being amazed at what you can do with this language

**Review by jaylees14 — APPROVED** [review]
Looks good :)

File: src/test/scala/temple/DSL/semantics/ValidatorTest.scala (hunk 1 above)
Comment (Lewiky) [reply src/test/scala/temple/DSL/semantics/ValidatorTest.scala:236]: scala best language

**Review by Lewiky — APPROVED** [review]
Sure

**Pull request merged**